imap_processing/ialirt/tests/unit/test_decom_ialirt.py

```diff
@@ -0,0 +1,197 @@
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from imap_processing import imap_module_directory
+from imap_processing.decom import decom_packets
+from imap_processing.ialirt.l0.decom_ialirt import generate_xarray
+
+IALIRT_PACKET_LENGTH = 1464
+
+
+@pytest.fixture(scope="session")
+def xtce_ialirt_path():
+    """Returns the xtce auxiliary directory."""
+    return imap_module_directory / "ialirt" / "packet_definitions" / "ialirt.xml"
+
+
+@pytest.fixture(scope="session")
+def binary_packet_path():
+    """
+    Creates a binary file from the text packet data, which is more representative
+    of the actual operational environment. The binary file is deleted after the
+    test session.
+    """
+    packet_path = (
+        imap_module_directory
+        / "ialirt"
+        / "tests"
+        / "test_data"
+        / "l0"
+        / "IALiRT Raw Packet Telemetry.txt"
+    )
+
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    with open(packet_path) as text_file, open(binary_file_path, "wb") as binary_file:
+        for line in text_file:
+            if not line.startswith("#"):
+                # Split the line by semicolons
+                # Discard the first value since it is only a counter
+                hex_values = line.strip().split(";")[1:]
+                # Convert hex to binary
+                binary_data = bytearray.fromhex("".join(hex_values))
+                binary_file.write(binary_data)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
+
+
+@pytest.fixture(scope="session")
+def empty_binary_packet_path():
+    """
+    Creates an empty binary file, representative of a scenario
+    in which no packet data is available.
+    """
+    with tempfile.NamedTemporaryFile(delete=False) as tmp_file:
+        binary_file_path = Path(tmp_file.name)
+
+    yield binary_file_path
+
+    binary_file_path.unlink()
+
+
+@pytest.fixture(scope="session")
+def decom_packets_data(binary_packet_path, xtce_ialirt_path):
+    """Read packet data from file using decom_packets"""
+    data_packet_list = decom_packets(binary_packet_path, xtce_ialirt_path)
+    return data_packet_list
+
+
+def test_length(decom_packets_data):
+    """Test if total packets in data file is correct"""
+    total_packets = 32
+    assert len(decom_packets_data) == total_packets
+
+
+def test_binary_value_length():
+    """
+    Validates the length of binary data converted
+    from a text file containing hexadecimal packet data.
```

Comment:
This function is sort of recreating what you're already doing in the fixture above. You could just add the assert up there if you're trying to test that file.